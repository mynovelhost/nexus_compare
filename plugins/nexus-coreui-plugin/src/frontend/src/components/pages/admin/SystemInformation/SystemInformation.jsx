/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React, {useState, useEffect} from 'react';

import {faDownload} from '@fortawesome/free-solid-svg-icons';

import Axios from 'axios';
import {BreadcrumbActions, Button, ContentBody, NxFontAwesomeIcon, NxLoadWrapper, Utils} from 'nexus-ui-plugin';
import UIStrings from '../../../../constants/UIStrings';

import SystemInformationBody from "./SystemInformationBody";

import './SystemInformation.scss';

const INITIAL_VALUE = {};

/**
 * @since 3.24
 */
export default function SystemInformation() {
  const [systemInformation, setSystemInformation] = useState(INITIAL_VALUE);
  const isLoaded = systemInformation !== INITIAL_VALUE;
  const isLoading = !isLoaded;

  useEffect(() => {
    if (isLoaded) {
      return;
    }

    Axios.get('/service/rest/atlas/system-information').then((result) => {
      setSystemInformation(result.data);
    });
  });

  function downloadSystemInformation() {
    window.open(Utils.urlFromPath('/service/rest/atlas/system-information'), '_blank');
  }

  return <>
    <BreadcrumbActions>
      <Button variant="primary" onClick={() => downloadSystemInformation()} disabled={!isLoaded}>
        <NxFontAwesomeIcon icon={faDownload}/>
        <span>{UIStrings.SYSTEM_INFORMATION.ACTIONS.download}</span>
      </Button>
    </BreadcrumbActions>
    <ContentBody className="nxrm-system-information">
      <NxLoadWrapper loading={isLoading}>
        <SystemInformationBody systemInformation={systemInformation} />
      </NxLoadWrapper>
    </ContentBody>
  </>;
}
