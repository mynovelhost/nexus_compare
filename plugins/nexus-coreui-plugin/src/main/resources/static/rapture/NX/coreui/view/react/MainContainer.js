/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
