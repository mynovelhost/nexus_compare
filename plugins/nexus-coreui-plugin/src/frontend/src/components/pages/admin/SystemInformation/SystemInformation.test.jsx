/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import {render, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SystemInformation from './SystemInformation';

jest.mock('axios', () => {  // Mock response from axios
  return {
    ...jest.requireActual('axios'), // Use most functions from actual axios
    get: jest.fn((url) => Promise.resolve({
      data: {
        'nexus-status': {
          'status': 'value'
        },
        'nexus-node': {
          'node': 0
        },
        'nexus-configuration': {
          'enabled': true
        },
        'nexus-properties': {
          'property': false
        },
        'nexus-license': {
          'fingerprint': 'hash'
        },
        'nexus-bundles': {
          '0': {
            'bundleId': 0,
            'enabled': true,
            'config': 'value'
          }
        },
        'system-time': {
          'time': 'value'
        },
        'system-properties': {
          'property': 'value'
        },
        'system-environment': {
          'property': 'value'
        },
        'system-runtime': {
          'property': 'value'
        },
        'system-network': {
          'en0': {
            'enabled': true
          }
        },
        'system-filestores': {
          '/dev/null': {
            'path': '/dev/null'
          }
        }
      }
    }))
  };
});

describe('SystemInformation', () => {
  it('renders correctly', async () => {
    const {container, queryByText} = render(<SystemInformation/>);

    expect(container).toMatchSnapshot();

    await wait(() => expect(container).toContainElement(queryByText('system-filestores')));

    expect(container).toMatchSnapshot();
  });
});
