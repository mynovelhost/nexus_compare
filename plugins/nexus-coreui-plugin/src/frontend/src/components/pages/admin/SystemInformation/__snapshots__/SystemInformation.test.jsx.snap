// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemInformation renders correctly 1`] = `
<div>
  <div
    class="nxrm-breadcrumb--actions"
  >
    <button
      class="nx-btn nx-btn--primary"
      disabled=""
      type="input"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-download fa-w-16 nx-icon"
        data-icon="download"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
          fill="currentColor"
        />
      </svg>
      <span>
        Download as JSON
      </span>
    </button>
  </div>
  <div
    class="nxrm-content-body nxrm-system-information"
  >
    <div
      class="nx-loading-spinner"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-notch fa-w-16 fa-spin nx-loading-spinner__icon nx-icon"
        data-icon="circle-notch"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
          fill="currentColor"
        />
      </svg>
      <span>
        Loadingâ€¦
      </span>
    </div>
  </div>
</div>
`;

exports[`SystemInformation renders correctly 2`] = `
<div>
  <div
    class="nxrm-breadcrumb--actions"
  >
    <button
      class="nx-btn nx-btn--primary"
      type="input"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-download fa-w-16 nx-icon"
        data-icon="download"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
          fill="currentColor"
        />
      </svg>
      <span>
        Download as JSON
      </span>
    </button>
  </div>
  <div
    class="nxrm-content-body nxrm-system-information"
  >
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        nexus-status
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              status
            </td>
            <td
              class="nxrm-information--value"
            >
              value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        nexus-node
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              node
            </td>
            <td
              class="nxrm-information--value"
            >
              0
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        nexus-configuration
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              enabled
            </td>
            <td
              class="nxrm-information--value"
            >
              true
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        nexus-properties
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              property
            </td>
            <td
              class="nxrm-information--value"
            >
              false
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        nexus-license
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              fingerprint
            </td>
            <td
              class="nxrm-information--value"
            >
              hash
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        nexus-bundles
      </h2>
      <div>
        <h3>
          0
        </h3>
        <table
          class="nxrm-information"
        >
          <tbody>
            <tr
              class="nxrm-information--row"
            >
              <td
                class="nxrm-information--name"
              >
                bundleId
              </td>
              <td
                class="nxrm-information--value"
              >
                0
              </td>
            </tr>
            <tr
              class="nxrm-information--row"
            >
              <td
                class="nxrm-information--name"
              >
                enabled
              </td>
              <td
                class="nxrm-information--value"
              >
                true
              </td>
            </tr>
            <tr
              class="nxrm-information--row"
            >
              <td
                class="nxrm-information--name"
              >
                config
              </td>
              <td
                class="nxrm-information--value"
              >
                value
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        system-time
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              time
            </td>
            <td
              class="nxrm-information--value"
            >
              value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        system-properties
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              property
            </td>
            <td
              class="nxrm-information--value"
            >
              value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        system-environment
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              property
            </td>
            <td
              class="nxrm-information--value"
            >
              value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        system-runtime
      </h2>
      <table
        class="nxrm-information"
      >
        <tbody>
          <tr
            class="nxrm-information--row"
          >
            <td
              class="nxrm-information--name"
            >
              property
            </td>
            <td
              class="nxrm-information--value"
            >
              value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        system-network
      </h2>
      <div>
        <h3>
          en0
        </h3>
        <table
          class="nxrm-information"
        >
          <tbody>
            <tr
              class="nxrm-information--row"
            >
              <td
                class="nxrm-information--name"
              >
                enabled
              </td>
              <td
                class="nxrm-information--value"
              >
                true
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="nxrm-section nx-tile"
    >
      <h2>
        system-filestores
      </h2>
      <div>
        <h3>
          /dev/null
        </h3>
        <table
          class="nxrm-information"
        >
          <tbody>
            <tr
              class="nxrm-information--row"
            >
              <td
                class="nxrm-information--name"
              >
                path
              </td>
              <td
                class="nxrm-information--value"
              >
                /dev/null
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
